pkg/jobrunaggregator/jobrunbigqueryloader/uploader.go

```diff
@@ -0,0 +1,275 @@
+package jobrunbigqueryloader
+
+import (
+	"context"
+	"fmt"
+	"sync"
+
+	"github.com/openshift/ci-tools/pkg/jobrunaggregator/jobrunaggregatorapi"
+	"github.com/openshift/ci-tools/pkg/jobrunaggregator/jobrunaggregatorlib"
+	"golang.org/x/sync/semaphore"
+	utilerrors "k8s.io/apimachinery/pkg/util/errors"
+	prowv1 "k8s.io/test-infra/prow/apis/prowjobs/v1"
+)
+
+type getLastJobRunWithDataFunc func(ctx context.Context, jobName string) (*jobrunaggregatorapi.JobRunRow, error)
+type shouldCollectDataForJobFunc func(job jobrunaggregatorapi.JobRow) bool
+
+func wantsTestRunData(job jobrunaggregatorapi.JobRow) bool {
+	return job.CollectTestRuns
+}
+func wantsDisruptionData(job jobrunaggregatorapi.JobRow) bool {
+	return job.CollectDisruption
+}
+
+type jobsBigQueryLoaderOptions struct {
```

Comment:
jobsBigQueryLoaderOptions is a little disorienting in proximity to jobBigQueryLoaderOptions. Suggest allJobsBigQueryLoaderOptions for readability. If this is ok let me know I'm happy to make the change.